internal/http/certificate.go

```diff
@@ -0,0 +1,22 @@
+package http
+
+import (
+	"crypto/x509"
+	"errors"
+	"os"
+)
+
+func LoadRootCAs(path string) (*x509.CertPool, error) {
+	pool, err := x509.SystemCertPool()
```

Comment:
The motivation of having user specified root CA pool is that the user cert is not in the system cert pool or the system cert pool is not trusted for this particular operation / scenario.

Therefore, we should start with a new cert pool instead.